clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -207,9 +207,10 @@ public synchronized void ensureCoordinatorReady() {
     }
 
     protected RequestFuture<Void> lookupCoordinator() {
-        if (findCoordinatorFuture == null) {
-            findCoordinatorFuture = sendGroupCoordinatorRequest();
-            findCoordinatorFuture.addListener(new RequestFutureListener<Void>() {
+        RequestFuture<Void> future = findCoordinatorFuture;
+        if (future == null) {
+            findCoordinatorFuture = future = sendGroupCoordinatorRequest();
```

Comment:
Ouch. Can I blame the reviewers for this? As an alternative, suppose we just synchronized lookupCoordinator() and removed the listener? For example:
```java
    protected synchronized RequestFuture<Void> lookupCoordinator() {
        if (findCoordinatorFuture == null || findCoordinatorFuture.isDone())
            findCoordinatorFuture = sendGroupCoordinatorRequest();
        return findCoordinatorFuture;
    }
```
Maybe that would be simpler?